states/huntsman/calibrating.py

```diff
@@ -4,116 +4,87 @@
-    """
-    Wait for the altitude of the Sun to be within given limits
-
-    Args:
-        pocs :
-        min_altitude (astropy.units.Quantity, optional):
-        max_altitude (astropy.units.Quantity, optional):
-        message (str):
-    """
-    if min_altitude is None and max_altitude is None:
-        raise ValueError("At least one of min_altitude & max_altitude must be given")
-    if min_altitude is None:
-        min_altitude = -90
-    if max_altitude is None:
-        max_altitude = 90
-    if isinstance(min_altitude, u.Quantity):
-        min_altitude = min_altitude.to(u.degree).value
-    if isinstance(max_altitude, u.Quantity):
-        max_altitude = max_altitude.to(u.degree).value
-
-    if not delay:
-        delay = pocs._safe_delay
-
-    while pocs.is_safe():
-        sun_pos = pocs.observatory.observer.altaz(current_time(),
-                                                  target=get_sun(current_time())
-                                                  ).alt
-        if sun_pos.value > max_altitude or sun_pos.value < min_altitude:
-            # Check simulator for 'night'
-            if 'night' in pocs.config['simulator']:
-                pocs.logger.info(f'Using night simulator, pretending sun down.')
-                break
-            else:
-                pocs.say(message.format(sun_pos.value))
-                pocs.sleep(delay=delay)
+def _wait_for_twilight(pocs, horizon):
+    '''
+    Wait for twighlight if safe to do so.
+    '''
+    delay = pocs._safe_delay
+
+    while pocs.is_safe(horizon='flat'):
+
+        if not pocs.is_dark(horizon=horizon):
+            pocs.say('Not dark enough for coarse focusing. Waiting...')
+            pocs.sleep(delay=delay)
         else:
-            break
+            return True
 
+    return False
 
-def on_enter(event_data):
-    """Pointing State
 
-    Take 30 second exposure and plate-solve to get the pointing error
-    """
-    pocs = event_data.model
+def _get_cameras(pocs):
+    '''
+    Get lists of narrow and broad band cameras.
+    '''
+    narrow_band_cameras = list()
+    broad_band_cameras = list()
+    for cam_name, cam in pocs.observatory.cameras.items():
+
+        # This is a hack. There should be a narrow property in the config...
+        if cam.filter_type.lower().startswith('ha'):
+            narrow_band_cameras.append(cam_name)
+        else:
+            broad_band_cameras.append(cam_name)
+
+    return narrow_band_cameras, broad_band_cameras
+
 
+def on_enter(event_data):
+    '''
+    Calibrating state. If safe to do so, take flats and darks. Should be
+    called once at the beginning and end of the night.
+    '''
+    pocs = event_data.model
     pocs.next_state = 'parking'
 
-    try:
+    # Leave the state if not safe to take calibrations
+    if not pocs.is_safe(horizon='flat'):
+        print('Exiting calibrating state because it is no longer safe.')
+        return
```

Comment:
This check shouldn't be necessary here as safety would have bee required to enter the state and nothing else has been done at this point. Should be able to just remove all this.

```suggestion
```